lib/pure/asyncdispatch.nim

```diff
@@ -1185,7 +1185,7 @@ else:
     var curList: seq[Callback]
 
     let selector = getGlobalDispatcher().selector
-    withData(selector, fd.int, fdData):
+    withData(selector, fd.int, fdData) do:
```

Comment:
This seems horrible, why are we making these templates that take a single body now require a `do`?

Reply:
No this `do` is not required. I've just put it here, because every other call to `withData` in this file uses `do`.